go/tools/bazel/runfiles.go

```diff
@@ -51,10 +51,21 @@ func Runfile(path string) (string, error) {
 	}
 
 	// Search manifest if we have one.
-	if entry, ok := runfiles.index[path]; ok {
```

Comment:
Are you sure all of the changes to this file are needed? I patched the change locally and the tests still passed without the changes to this file.

Reply:
Yeah, it's necessary on Windows - the way this works on non-Windows is that first we'll stat the file, see the symlink and be happy.

But the way it works on Windows is we parse a metadata file with paths to files written down, and we do a match based on exactly the file name inside whatever workspace the file happens to be in. These changes are necessary for that runfiles lookup to work properly.

I could, if you'd prefer, make the test use a file in the main workspace (rather than from the `go_sdk` workspace), and send a separate PR fixing up the separate-workspace runfiles lookup issue, but figured fixing it all in one was reasonable...

Reply:
That makes sense, thanks for the clarification. Yeah, I think including the change as part of this PR is reasonable.